### PR TITLE
Add regex validation for custom attributes in the self registration

### PR DESCRIPTION
**Pull request by sumedhe**
### Purpose
Add regex validation for custom attributes in the self registration.

### Related Issues
- https://github.com/wso2/product-is/issues/17918

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-username-request.jsp [hunk 1]
```diff
@@ -1806,6 +1806,18 @@
 
                 return false;
             }
+            if (element.type === 'text' && element.value != null && !element.checkValidity()) {
+                $("#" + error_msg_txt).text("<%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
+                    "Please.enter.valid.input")%>");
+                $("#" + error_msg_element).show();
+                $("#" + element_field).addClass("error");
+                var error_msg_txt_element = document.getElementById(error_msg_txt);
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-username-request.jsp:1814]: Shouldn't hash be appended here?

```suggestion
                var error_msg_txt_element = document.getElementById("#" + error_msg_txt);
```

**Comment by dasuni-30** [comment]
Let's add the changeset

File: identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_zh_CN.properties [hunk 2]
```diff
@@ -139,6 +139,7 @@ Invalid.reset.link=此链接已过期。
 Set.Password=设置密码
 Continue=继续
 Please.enter.valid.email=请输入有效的电子邮件地址。
+Please.enter.valid.input=请输入有效的条目。
```
Comment (Achintha444) [inline identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_zh_CN.properties:142]: chinese and japanses does not seems to work when we use with idea IDE. Hence try it using vs code.

File: identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-username-request.jsp (hunk 1 above)
Comment (sumedhe) [reply identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-username-request.jsp:1814]: No need to put `#` prefix with `getElementById()`.

File: identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_zh_CN.properties (hunk 2 above)
Comment (sumedhe) [reply identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_zh_CN.properties:142]: Used vscode to put these lines. Anything to change?

Comment (Achintha444) [reply identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_zh_CN.properties:142]: Ack

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**